From 291dbf3d63d8b591a9255853858e2540238a8ff0 Mon Sep 17 00:00:00 2001
From: Eduardo Otubo <otubo@redhat.com>
Date: Mon, 27 Apr 2020 10:52:52 +0200
Subject: [PATCH 1/2] Remove race condition between cloud-init and
 NetworkManager

RH-Author: Eduardo Otubo <otubo@redhat.com>
Message-id: <20200327121911.17699-1-otubo@redhat.com>
Patchwork-id: 94453
O-Subject: [RHEL-7.9/RHEL-8.2.0 cloud-init PATCHv2] Remove race condition between cloud-init and NetworkManager
Bugzilla: 1748015
RH-Acked-by: Cathy Avery <cavery@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

cloud-init service is set to start before NetworkManager service starts,
but this does not avoid a race condition between them. NetworkManager
starts before cloud-init can write `dns=none' to the file:
/etc/NetworkManager/conf.d/99-cloud-init.conf. This way NetworkManager
doesn't read the configuration and erases all resolv.conf values upon
shutdown. On the next reboot neither cloud-init or NetworkManager will
write anything to resolv.conf, leaving it blank.

This patch introduces a NM reload (try-reload-or-restart) at the end of cloud-init
start up so it won't erase resolv.conf upon first shutdown.

x-downstream-only: yes

Signed-off-by: Eduardo Otubo otubo@redhat.com
---
 rhel/systemd/cloud-final.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rhel/systemd/cloud-final.service b/rhel/systemd/cloud-final.service
index f303483..8510520 100644
--- a/rhel/systemd/cloud-final.service
+++ b/rhel/systemd/cloud-final.service
@@ -12,7 +12,7 @@ RemainAfterExit=yes
 TimeoutSec=0
 KillMode=process
 ExecStartPost=/bin/echo "try restart NetworkManager.service"
-ExecStartPost=/usr/bin/systemctl try-restart NetworkManager.service
+ExecStartPost=/usr/bin/systemctl try-reload-or-restart NetworkManager.service
 
 # Output needs to appear in instance console output
 StandardOutput=journal+console
-- 
1.8.3.1

